crates/rolldown/src/ast_scanner/impl_visit.rs

```diff
@@ -60,48 +60,6 @@ impl<'me, 'ast: 'me> Visit<'ast> for AstScanner<'me, 'ast> {
     }
   }
 
-  fn visit_member_expression(&mut self, expr: &MemberExpression<'ast>) {
-    match expr {
-      MemberExpression::StaticMemberExpression(member_expr) => {
-        // For member expression like `a.b.c.d`, we will first enter the (object: `a.b.c`, property: `d`) expression.
-        // So we add these properties with order `d`, `c`, `b`.
-        let mut props_in_reverse_order = vec![];
-        let mut cur_member_expr = member_expr;
-        let object_symbol_in_root_scope = loop {
-          props_in_reverse_order.push(&cur_member_expr.property);
-          match &cur_member_expr.object {
-            Expression::StaticMemberExpression(expr) => {
-              cur_member_expr = expr;
-            }
-            Expression::Identifier(id) => {
-              break self.resolve_identifier_to_root_symbol(id);
-            }
-            _ => break None,
-          }
-        };
-        match object_symbol_in_root_scope {
-          // - Import statements are hoisted to the top of the module, so in this time being, all imports are scanned.
-          // - Having empty span will also results to bailout since we rely on span to identify ast nodes.
-          Some(sym_ref)
-            if self.result.named_imports.contains_key(&sym_ref) && !expr.span().is_unspanned() =>
-          {
-            let props = props_in_reverse_order
-              .into_iter()
-              .rev()
-              .map(|ident| ident.name.as_str().into())
-              .collect::<Vec<_>>();
-            self.add_member_expr_reference(sym_ref, props, expr.span());
-            // Don't walk again, otherwise we will add the `object_symbol_in_root_scope` again in `visit_identifier_reference`
-            return;
```

Comment:
This early return leads visitor can't reach `visit_identifier_reference`, which may need duplicate visitor logic.